eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentLifecycle.scala

```diff
@@ -45,6 +45,12 @@ class PaymentLifecycle(nodeParams: NodeParams, id: UUID, router: ActorRef, regis
   startWith(WAITING_FOR_REQUEST, WaitingForRequest)
 
   when(WAITING_FOR_REQUEST) {
+    case Event(SendPaymentToRoute(amountMsat, paymentHash, hops, finalCltvExpiry), WaitingForRequest) =>
```

Comment:
nit: that's not consistent with the handler style below (I found that argument extraction is less maintenable, everytime you add an arg they don't compile).

Reply:
I'm okay with this proposed change even though in general i like the compiler to do the heavy lifting and error-catching, but in the case of the extractor pattern the compile time errors are typically not very useful, addressed in 52e4938.